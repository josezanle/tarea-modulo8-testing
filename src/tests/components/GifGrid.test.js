import React from 'react';
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('pruebas en el componente <GifGrid />', () => {

    const category = 'one punch';

    test('debe mostrarse corretamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading:true
        });
        
        const wrapper = shallow (<GifGrid category={category} /> );
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {

        const gifs = [
        
        {
            id:'ABC',
            url:'https://localhost/cualquier/cosa.jpg',
            title:'lo que sea'
        },

        {
            id: '123',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'lo que sea'
        }
    ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category} />);

        // expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
        
    })
    
    
    
});
