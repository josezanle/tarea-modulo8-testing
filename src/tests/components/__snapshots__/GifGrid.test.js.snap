// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente <GifGrid /> debe mostrarse corretamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    one punch
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
